### PR TITLE
SI-7649 Fix positions for reshaped tag materializers

### PR DESCRIPTION
**Pull request by retronym**
Calls to `materializeClassTag[T]` are replaced during reification with
`implicitly[ClassTag[T]]` in the `reify` macro. This is done to avoid
referring to symbols in scala-compiler.jar. Class- and Type-Tag
materialization is treated in the same way.

This change positions the replacement trees to avoid triggering
assertions under -Yrangepos.

Review by @xeno-by.

Are we in a better shape to address this problem properly in master?

``` scala
// when implicit macros are fixed, these sneaky macros will move to corresponding companion objects
// of, say, ClassTag or TypeTag
```

### CODE REVIEW

File: src/compiler/scala/reflect/reify/phases/Reshape.scala [hunk 1]
```diff
@@ -91,20 +91,20 @@ trait Reshape {
     private def undoMacroExpansion(tree: Tree): Tree =
       tree.attachments.get[MacroExpansionAttachment] match {
         case Some(MacroExpansionAttachment(original)) =>
+          def mkImplicitly(tp: Type) = atPos(tree.pos.makeTransparent)(
```
Comment (xeno-by) [inline src/compiler/scala/reflect/reify/phases/Reshape.scala]: Why make the position transparent?

**Comment by xeno-by** [comment]
I think that indeed we can redo tag materialization with implicit macros.

File: src/compiler/scala/reflect/reify/phases/Reshape.scala (hunk 1 above)
Comment (retronym) [inline src/compiler/scala/reflect/reify/phases/Reshape.scala]: Actually it's not needed, I've just removed it.

**Comment by xeno-by** [comment]
Well then LGTM

**Pull request merged**